lib/_stream_writable.js

```diff
@@ -155,11 +155,12 @@ Writable.prototype.pipe = function() {
 };
 
 
-function writeAfterEnd(stream, cb) {
-  var er = new Error('write after end');
-  // TODO: defer error events consistently everywhere, not just the cb
-  stream.emit('error', er);
-  process.nextTick(cb, er);
+function writeAfterEndErr(stream, cb) {
+  const err = new Error('write after end');
+  process.nextTick(() => {
+    stream.emit('error', err);
+    cb(err);
+  });
```

Comment:
the readable-stream module is build from the node source using a couple of build methods, https://github.com/nodejs/readable-stream/tree/master/build to get better browser support.